soccer_vision_msgs/msg/MarkingIntersection.msg

```diff
@@ -0,0 +1,29 @@
+# Describes a detected marking intersection from vision modules, in image coordinates.
+# An intersection is defined as a point where two line markings intersect.
+# The term ray is used to describe the outgoing rays from the center of the
+# intersection.
+#
+# For example, looking down at a perfectly upright T-intersection on the ground from above, you would have:
+# - num_rays = 3
+# - heading_rays = [0, pi / 2, pi]
+
+float32 CONFIDENCE_UNKNOWN=-1
+
+# Center point of the intersection, the z-axis should be ignored (in pixel)
+geometry_msgs/Point center
+
+# The number of rays outgoing from an intersection. This is 3 for a T-Junction and 
+# 4 for an X-Junction
+int32 num_rays
```

Comment:
Also, why do we need to specify this? I'd assume, you can infer the count of rays from the array below. If we want to specify what type of intersection it is, I think we should define it explicitly.

Reply:
I don't think L, T and X covers all cases. In my mind, an L, T and X intersection isn't just intersections with varying number of rays merging, but they also have to meet at 90 degrees when seen from above.

This is the case where an intersection can exist, but I wouldn't classify it as a T intersection. Rather, it makes sense to say there are 3 rays.
![Screenshot from 2021-10-06 09-27-17](https://user-images.githubusercontent.com/18237662/136121936-6788b0f5-68c0-4cd3-856c-5eab3f872b81.png)